File: apps/order-management/src/APIs/Customer/Dtos/CustomerUpdateInput.cs
namespace OrderManagementDotNet.APIs.Dtos;

public class CustomerUpdateInput
{
    public string? Id { get; set; }

    public DateTime? CreatedAt { get; set; }

    public string? UpdatedAt { get; set; }

    public string? FirstName { get; set; }

    public string? LastName { get; set; }

    public string? Email { get; set; }

    public string? Phone { get; set; }

    public List<OrderIdDto>? Orders { get; set; }

    public AddressIdDto? Address { get; set; }
}

File: apps/custom-2-admin/src/api/customer/CustomerUpdateInput.ts
import { AddressWhereUniqueInput } from "../address/AddressWhereUniqueInput";
import { MorUpdateManyWithoutCustomersInput } from "./MorUpdateManyWithoutCustomersInput";
import { OrderUpdateManyWithoutCustomersInput } from "./OrderUpdateManyWithoutCustomersInput";

export type CustomerUpdateInput = {
  address?: AddressWhereUniqueInput | null;
  email?: string | null;
  firstName?: string | null;
  lastName?: string | null;
  mors?: MorUpdateManyWithoutCustomersInput;
  orders?: OrderUpdateManyWithoutCustomersInput;
  phone?: string | null;
};
